source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc

```diff
@@ -51,35 +51,21 @@ void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckRe
     span.setTag(TracingConstants::get().TraceStatus, TracingConstants::get().TraceOk);
     authz_response->status = CheckStatus::OK;
     if (response->has_ok_response()) {
-      toAuthzResponseHeader(authz_response, response->ok_response().headers());
-      if (response->ok_response().headers_to_remove_size() > 0) {
-        for (const auto& header : response->ok_response().headers_to_remove()) {
-          authz_response->headers_to_remove.push_back(Http::LowerCaseString(header));
-        }
-      }
-      if (response->ok_response().query_parameters_to_set_size() > 0) {
-        for (const auto& query_parameter : response->ok_response().query_parameters_to_set()) {
+      auto ok_response = response->ok_response();
+      copyHeaderMutations(authz_response, ok_response.headers(),
```

Comment:
Is it simpler to pass a reference of ok_response as multiple parameters are within it?

Reply:
I've written it so that it uses the same function for copying denied_response and ok_response header mutations. denied_response does not use all the same fields.